### DISCUSSION TITLE
Static Media Failure

### DISCUSSION THREAD

**Discussion by Quembar** [netbox-community/netbox#5856] [Q&A] [answered]
During Installation I came to chapter 5 'HTTP Server'. Until now I followed the installation guide and have only less issues (I have to do a look for the redis service. But this is not relevant in this case).

Now I get the message "Static Media Failure" when call the URL https://<IP or FQDN of the NetBox-Server>

I have checked the three hints. The files are at the place and they are reachable.
[edit] But apache error log shows: "AH00037: Symbolic link not allowed or link target not accessible: /opt/netbox,"[/edit]

What should I check next? [edit] I assume that there is a missing entry in the vhost config.[/edit]

(OS is openSUSE 15.2; Apache2)

**Comment by jhammond-git** [comment]
A cursory google check implies this may be related to directory or symlink permissions.  Make sure that the user that apache is running as has permission to be in /opt, and the directory that /opt/netbox points to.  Also that the symlink is owned by the same user as the target directory.

**Reply by Quembar** [reply]
Yes, I found a lot of such hints about apache at google by myself ...
But even if the user wwwrun:www (this is the Apache user and group) own all files and folders in /opt the message is the same.
I run some instance on apache on other hosts, but this is the first with such an structure python, django and so on. So i ask for help here.

**Comment by candlerb** [comment, marked as answer]
The short answer is, there's an error in your Apache config.  However without seeing your complete Apache config, it's hard to be more specific.

It's also unclear which version of installation instructions you followed.  Did you [download and unpack a release archive](https://netbox.readthedocs.io/en/stable/installation/3-netbox/#option-a-download-a-release-archive), or did you [clone the git repository](https://netbox.readthedocs.io/en/stable/installation/3-netbox/#option-b-clone-the-git-repository)?

If you chose the former option, then the instructions recommend creating a symlink from `/opt/netbox` to `/opt/netbox-x.x.x`.  Did you do that?  It may be relevant, given that the error is about symlinks.

Did you use the supplied [`/opt/netbox/contrib/apache.conf`](https://github.com/netbox-community/netbox/blob/develop/contrib/apache.conf) ?  It includes a setting which relates to symlinks:

```
    <Directory /opt/netbox/netbox/static>
        Options Indexes FollowSymLinks MultiViews
        AllowOverride None
        Require all granted
    </Directory>
```

Now, the other thing is you're using OpenSUSE.  The default `/etc/apache2/apache2.conf` in Ubuntu includes the following:

```
<Directory />
        Options FollowSymLinks
        AllowOverride None
        Require all denied
</Directory>
```

Maybe OpenSUSE doesn't have that?

**Reply by Quembar** [reply]
Yes, probable a error in the apache config. I assume it. The symlink is done. All file owner are checked. The supplied apache config is used.
But the difference between Ubuntu and openSUSE in the default config of the apache did the trick. After I add the Ubuntu part you mentioned in the default-server.conf the error is gone.
Thank you for this hint!

<Edit>
It seems better to at the Ubuntu part into the supplied apache.conf and store this as vhost entry in openSUSE.
```

Listen 443
<VirtualHost *:443>
    ProxyPreserveHost On

    # CHANGE THIS TO YOUR SERVER'S NAME
    ServerName netbox.example.com

    SSLEngine on
    SSLCertificateFile /etc/ssl/certs/netbox.crt
    SSLCertificateKeyFile /etc/ssl/private/netbox.key

    Alias /static /opt/netbox/netbox/static

    <Directory /opt/netbox/netbox/static>
        Options Indexes FollowSymLinks MultiViews
        AllowOverride None
        Require all granted
    </Directory>

    <Directory />
        Options FollowSymLinks
        AllowOverride None
        Require all denied
    </Directory>

    <Location /static>
        ProxyPass !
    </Location>

    RequestHeader set "X-Forwarded-Proto" expr=%{REQUEST_SCHEME}
    ProxyPass / http://127.0.0.1:8001/
    ProxyPassReverse / http://127.0.0.1:8001/
</VirtualHost>
```
Maybe someone wants to update the documentation.

</Edit>
